src/client/cockpit-client

```diff
@@ -114,6 +115,55 @@ class CockpitClientWindow(Gtk.ApplicationWindow):
     def open_inspector(self, *unused):
         self.webview.get_inspector().show()
 
+    def run_js(self, action, parameter, *unused):
```

Comment:
So here's my counter-proposal:

 - add a `--enable-run-js` commandline argument which sets `self.enable_run_js = True` on the app
 - check `app.enable_run_js` when constructing the window and in case it's set:
   - install the action
   - register the script message handler → it will just always set the action state
   - register the page-load handler → it will just always set the action state
   - never unregister anything

or is it actually required to register the handler each time separately because the JS context got destroyed in between?

Reply:
I added the `--enable-run-js` hidden option. I'll look into keeping the handler next week. I am rather sure the context gets destroyed between page loads (that's why I have to split up the test so heavily in the first place), but I'll experiment with that -- would certainly make things easier.

Reply:
I pushed a fixup which implements your suggestion. Honestly I don't like it that much, as it smears out the run-js functionality across the code, and it is overall an extra three lines of code. Also, it does not really simplify things other than the two lines of signal handler cleanup -- but that also makes the action state a lot more muddy and potentially racy.

It does work to leave the handlers permanently, though.

Reply:
Okay.  After looking at the diff, I agree that it's not awesome.  Let's do it your way.

But, one detail that might make this seem a bit nicer to me:

install the `run-js` action unconditionally at the same time as all of the others, but directly under the call to `add_action_entries()` put something like:

`self.lookup_action('run-js').set_enabled(app.enable_run_js)`

Reply:
Done, thanks for the hint!